packages/scroll-sync/src/ScrollSyncPane.tsx

```diff
@@ -0,0 +1,47 @@
+import {
+	Children,
+	cloneElement,
+	forwardRef,
+	isValidElement,
+	type ReactElement,
+	type RefAttributes,
+	useEffect,
+	useRef,
+	FunctionComponentElement,
+	ForwardedRef,
+} from 'react';
+import { useScrollSync } from './ScrollSyncContext';
+import { useMergeRefs } from './useMergeRefs';
+
+export interface IScrollSyncPaneProps {
+	children: ReactElement<RefAttributes<unknown>>;
+}
+
+export const ScrollSyncPane = forwardRef<unknown, IScrollSyncPaneProps>(function ScrollSyncPaneRef({ children }, ref) {
+	const childRef = useRef<HTMLElement | null>(null);
+	const { registerPane, unregisterPane } = useScrollSync();
+
+	useEffect(() => {
+		registerPane(childRef);
+
+		return () => {
+			unregisterPane(childRef);
+		};
+	}, [registerPane, unregisterPane]);
```

Comment:
I have an idea how to avoid useEffect.
`ref` is called with `null` on unmount, you can use that in combination with `useId`.

```tsx
const id = useId();

const mergedRefs = useMergeRefs(
		...,
		(node) => {
		   if (node === null) {
		     unregisterPane(id);
		   }
		   registerPane(id, node);
		}
	);
```
But you'll need to change `unregisterPane` implementation.

Reply:
I have implemented it like this and it seems that it works! also, I have checked if the `unregister` is being called, and it is.

Reply:
This is just an idea to avoid useLayouEffect. Ref callback will be called before useLayoutEffect but you don’t need to handle SSR.